grammar.js

```diff
@@ -644,11 +636,61 @@ module.exports = grammar({
 
     operator_identifier: $ => /[^\s\w\(\)\[\]\{\}'"`\.;,]+/,
 
-    number: $ => /[\d\.]+/,
+    literal: $ => choice(
+      $.integer_literal,
+      $.floating_point_literal,
+      $.boolean_literal,
+      $.character_literal,
+      $.symbol_literal,
+      $.string
+    ),
+
+    integer_literal: $ => token(
+      seq(
+        optional(/[-]/),
+        choice(
+          /[\d]+/,
+          /0[xX][\da-fA-F]+/
+        ),
+        optional(/[lL]/)
+      )
+    ),
+
+    floating_point_literal: $ => token(
+      seq(
+        optional(/[-]/),
+        choice(
+          // digit {digit} ‘.’ digit {digit} [exponentPart] [floatType]
+          seq(
+            /[\d]+\.[\d]+/,
+            optional(/[eE][+-]?[\d]+/),
+            optional(/[dfDF]/)
+          ),
+          // ‘.’ digit {digit} [exponentPart] [floatType]
+          seq(
+            /\.[\d]+/,
+            optional(/[eE][+-]?[\d]+/),
+            optional(/[dfDF]/)
+          ),
+          // digit {digit} exponentPart [floatType]
+          seq(
+            /[\d]+/,
+            /[eE][+-]?[\d]+/,
+            optional(/[dfDF]/)
+          ),
+          // digit {digit} [exponentPart] floatType
+          seq(
+            /[\d]+/,
+            optional(/[eE][+-]?[\d]+/),
+            /[dfDF]/
+          )
+        )
+      )
+    ),
 
     boolean_literal: $ => choice('true', 'false'),
 
-   character_literal: $ => token(seq(
+    character_literal: $ => token(seq(
```

Comment:
It looks like there's a stray whitespace added at the start of this line.

Reply:
This one is intentional to align indentation of `character_literal` to its neighbours.